frontend/src/components/guideCarousel/keyboardTemplate/index.js

```diff
@@ -0,0 +1,47 @@
+import { makeStyles } from 'tss-react/mui'
+import { useRef, useEffect } from 'react'
+import ShiftKeyboard from './shift'
+import NormalKeyboard from './normal'
+
+const useStyles = makeStyles()((theme) => { 
+
+  return {
+    root: {
+      '.active': {
+        '& rect': {
+          stroke: `${theme.palette.info.main} !important`
+        },
+        '& path': {
+          fill: `${theme.palette.info.main} !important`
+        }
+      },
+      '.current': {
+        '& rect': {
+          fill: `${theme.palette.info.main} !important`,
+          stroke: `${theme.palette.info.main} !important`
+        },
+        '& path': {
+          fill: `${theme.palette.common.white} !important`
+        }
+      },
+    },
+  }
+})
+
+const KeyboardTemplate = ({ template = 'normal', actions = [], current }) => {
+  const keyboardRef = useRef(null)
+  const Component = template === 'normal' ? NormalKeyboard : ShiftKeyboard
+  const { classes } = useStyles()
+
+  useEffect(() => {
+    actions
+      .filter(action => action !== current)
+      .forEach(action => keyboardRef.current.querySelector(`[data-action="${action}"]`)?.classList.add('active'))
+    
+    keyboardRef.current.querySelector(`[data-action="${current}"]`)?.classList.add('current')
```

Comment:
This line duplicates the one in forEach, the diff is only in 2 params.

Reply:
this part will be rewritten when animation for guide is ready

Reply:
this code will be rewritten when animation is ready